daemon/logger/loggerutils/logfile.go

```diff
@@ -641,26 +642,42 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 }
 
 func watchFile(name string) (filenotify.FileWatcher, error) {
-	fileWatcher, err := filenotify.New()
-	if err != nil {
-		return nil, err
-	}
 
 	logger := logrus.WithFields(logrus.Fields{
 		"module": "logger",
 		"fille":  name,
 	})
 
-	if err := fileWatcher.Add(name); err != nil {
+	var fileWatcher filenotify.FileWatcher
+
+	if runtime.GOOS == "windows" {
+		// FileWatcher on Windows files is based on the syscall notifications which has an issue becuase of file caching.
+		// It is based on ReadDirectoryChangesW() which doesn't detect writes to the cache. It detects writes to disk only.
+		// Becuase of the OS lazy writing, we don't get notifications for file writes and thereby the watcher
+		// doesn't work. Hence for Windows we will use poll based notifier.
+		fileWatcher = filenotify.NewPollingWatcher()
+	} else {
+		var err error
+		fileWatcher, err = filenotify.New()
+		if err != nil {
+			return nil, err
+		}
+
+		if err := fileWatcher.Add(name); err == nil {
```

Comment:
Can probably keep this check outside the if-block. Not particularly harmful to error out twice in the case of Windows.

Reply:
Not sure I followed this comment. For non Windows we retry with polling right?

Reply:
Wouldn't the fallback logic be duplicate for Windows?

Reply:
Yes, but doesn't really hurt anything.

Reply:
Ok. I updated the code as per your comment.